internal/operators/rx.go

```diff
@@ -6,44 +6,41 @@
 package operators
 
 import (
-	"fmt"
-
 	"github.com/corazawaf/coraza/v3/rules"
 
-	"github.com/corazawaf/coraza-proxy-wasm/internal/re2"
+	re2 "github.com/wasilibs/go-re2"
 )
 
 type rx struct {
-	re    re2.RegExp
-	debug bool
+	re *re2.Regexp
 }
 
 var _ rules.Operator = (*rx)(nil)
 
 func newRX(options rules.OperatorOptions) (rules.Operator, error) {
-	o := &rx{}
 	data := options.Arguments
 
-	if data == `(?:\$(?:\((?:\(.*\)|.*)\)|\{.*})|\/\w*\[!?.+\]|[<>]\(.*\))` {
-		o.debug = true
-		fmt.Println("enabling rx debug!")
-	}
-
 	re, err := re2.Compile(data)
 	if err != nil {
 		return nil, err
 	}
-
-	o.re = re
-	return o, err
+	return &rx{re: re}, nil
 }
 
 func (o *rx) Evaluate(tx rules.TransactionState, value string) bool {
-	res := o.re.FindStringSubmatch8(value, func(i int, match string) {
```

Comment:
With the GC improvements, this hyperoptimization is not needed anymore. The code is now copied from coraza, just changing the regexp import